source/common/stats/stats_matcher_impl.cc

```diff
@@ -89,21 +89,25 @@ bool StatsMatcherImpl::fastRejectMatch(StatName stat_name) const {
 }
 
 bool StatsMatcherImpl::slowRejects(FastResult fast_result, StatName stat_name) const {
-  bool match = slowRejectMatch(stat_name);
-
-  if (is_inclusive_ || match || fast_result != FastResult::Matches) {
-    //  is_inclusive_ | match | return
-    // ---------------+-------+--------
-    //        T       |   T   |   T   Default-inclusive and matching an (exclusion) matcher, deny.
-    //        T       |   F   |   F   Otherwise, allow.
-    //        F       |   T   |   F   Default-exclusive and matching an (inclusion) matcher, allow.
-    //        F       |   F   |   T   Otherwise, deny.
-    //
-    // This is an XNOR, which can be evaluated by checking for equality.
-    return match == is_inclusive_;
+  // Skip slowRejectMatch if we already have a definitive answer from fastRejects.
+  if (fast_result == FastResult::Rejects) {
```

Comment:
minor perf nit: You can write this with one conditional:
```
if (fast_result != FastResult::NoMatch) {
  return fast_result == FastResult::Rejects;
}
```

The 'fast rejects' case actually should not be reachable except via unit test.  Is that right? Cause in thread_local_store.cc we short-circuit and don't call this if we get a fast rejection.

Reply:
That is true. However, since these are public methods, I feel it's safer no to make assumptions on what checks happen outside this method.

We could also have an ASSERT to make sure fast_result is not `FastResult::Rejects` (I would need to remove some of the tests I've added with that change).

Which do you prefer?

Reply:
I think it should be sufficient to point out that in a comment, but keep the library robust to callers.

Right now handling this case which doesn't occur, causes an extra conditional. But if you implement my suggested change then the extra conditional will not be needed.